libs/model/src/community/GetMembers.query.ts

```diff
@@ -88,40 +78,25 @@ export function GetMembers(): Query<typeof schemas.GetCommunityMembers> {
       // This query is overly complex in order to entice the query planner to use the trigram indices
       const sqlWithoutPagination = `
     SELECT
-      "Profiles".id,
-      "Profiles".user_id,
-      "Profiles".profile_name,
-      "Profiles".avatar_url,
-      "Profiles".created_at,
-      array_agg("Addresses".id) as address_ids,
-      array_agg("Addresses".community_id) as community_ids,
-      array_agg("Addresses".address) as addresses,
-      MAX("Addresses".last_active) as last_active
-      FROM "Profiles"
-      JOIN "Addresses" ON "Profiles".user_id = "Addresses".user_id
-      WHERE ${communityWhere} 
-      ("Profiles".profile_name ILIKE '%' || :searchTerm || '%')
-      ${membershipsWhere}
-      GROUP BY "Profiles".id
-
-      UNION
-
-      SELECT
-      "Profiles".id,
-      "Profiles".user_id,
-      "Profiles".profile_name,
-      "Profiles".avatar_url,
-      "Profiles".created_at,
-      array_agg("Addresses".id) as address_ids,
-      array_agg("Addresses".community_id) as community_ids,
-      array_agg("Addresses".address) as addresses,
-      MAX("Addresses".last_active) as last_active
-      FROM "Profiles"
-      JOIN "Addresses" ON "Profiles".user_id = "Addresses".user_id
-      WHERE ${communityWhere} 
-      ("Addresses".address ILIKE '%' || :searchTerm || '%')
-      ${membershipsWhere}
-      GROUP BY "Profiles".id
+      U.id AS user_id,
+      U.profile->>'name' AS profile_name,
+      U.profile->>'avatar_url' AS avatar_url,
+      U.created_at,
+      array_agg(A.id) as address_ids,
+      array_agg(A.community_id) as community_ids,
+      array_agg(A.address) as addresses,
+      MAX(COALESCE(A.last_active, U.created_at)) as last_active
+      FROM "Users" U
+      JOIN "Addresses" A ON U.id = A.user_id
+      WHERE 
+        ${replacements.community_id ? 'A.community_id = :community_id' : ''} 
+        ${
+          replacements.searchTerm
+            ? "AND ((U.profile->>'name' ILIKE '%' || :searchTerm || '%') OR (A.address ILIKE '%' || :searchTerm || '%'))"
```

Comment:
Could you please include the query that you ran to get the query plan in https://explain.dalibo.com/ (paste it in) so we can review the specific case you are testing. The new query plan is using an index on `User.id` (for grouping) but then filtering over the profile name. We should be using an index for `ILIKE` search which in this case we are not. The index was recently added to the profiles table by @kurtisassad so we should not regress by not creating an index on `Users.profile.name`.

Reply:
@timolegros I'm doing a full refactoring of this query, simplifying the code, fixing some bugs in the UI, and removing the external calls to protocol helpers to get stake balances as part of the search - next step is to create an async process to project (backfill) stake balances to "Addresses" on a schedule and when stake is traded.

In the meantime this is the new query plan I'm working with:

https://explain.dalibo.com/plan/1c9gd7b7a6749597

Reply:
If you don't use stake helpers _and_ there is no refresh policy for the stake balances in Addresses you will end up using very stale balances and therefore incorrect ordering. TBC will refresh balances either when the cached value exceeds expiry or when an explicit refresh is requested. This means values are always relatively fresh. There is no way around having to fetch balances from TBC to get _fresh_ stake balances for ordering (can't correctly order on a partial or stale set). 

Having stake balances in Postgres would mean creating a separate refresh policy/job that refreshes balances by querying the TBC either before/after a request or periodically. Periodically is a bad idea since we would request balances when we don't need them and before (aka on) a request wouldn't be performant since you need to fetch from TBC and then save the data to thousands of rows in Postgres before returning a response. Refreshing Postgres balances after might be the best option but then you also risk stale balances unless you replicate TBC/Redis balance expiry in Postgres which would be cumbersome when the goal is just to order by stake balances.

The questions are:
1. How stale is too stale?
2. Are the performance gains of adding stake balances to Postgres (that meet the stale time requirements from question 1) worth the cost/work/complexity when compared to simply ordering in the route itself via a simple loop?